nep-11.mediawiki

```diff
@@ -0,0 +1,259 @@
+  Status: Accepted
+  Created: 2018-04-18
+</pre>
+
+==Abstract==
+
+This NEP defines a standard non-fungible token system for the NEO Smart Economy.
+
+==Motivation==
+
+NFTs are required to track, exchange and enforce ownership of digital assets.
+A non-fungible token (NFT) can be thought of like a property deed - each one is unique and carries some non-mutable information (e.g. the physical address of the property).
+Other information, such as the owner of the property, can be changed.
+Also, we provide built-in optional divisibility within each non-fungible asset.
+This allows for high value objects to be tokenized more effectively.
+
+==Specification==
+
+===Common methods===
+
+====symbol====
+
+<pre>
+public static string symbol()
+</pre>
+
+Returns a short string symbol of the token managed in this contract. e.g. <code>"MNFT"</code>.
+This symbol SHOULD be short (3-8 characters is recommended), with no whitespace characters or new-lines and SHOULD be limited to the uppercase latin alphabet (i.e. the 26 letters used in English).
+
+This method MUST always return the same value every time it is invoked.
+
+====decimals====
+
+<pre>
+public static byte decimals()
+</pre>
+
+Returns the number of decimals used by the token - e.g. <code>8</code>, means to divide the token amount by <code>100,000,000</code> to get its user representation.
+
+If the token managed in this contract is indivisible, the function SHOULD return <code>0</code>.
+
+If this method returns <code>0</code>, the "Non-divisible NFT methods" must be implemented.
+Otherwise, the "Divisible NFT methods" must be implemented.
+
+This method MUST always return the same value every time it is invoked.
+
+====totalSupply====
+
+<pre>
+public static BigInteger totalSupply()
+</pre>
+
+Returns the total token supply currently in the system.
+
+====balanceOf====
+
+<pre>
+public static BigInteger balanceOf(byte[] owner)
+</pre>
+
+Returns the total amount of NFTs owned by the specified address.
+
+The parameter <code>owner</code> SHOULD be a 20-byte address.
+If not, this method SHOULD <code>throw</code> an exception.
+
+====tokensOf====
+
+<pre>
+public static iterator tokensOf(byte[] owner)
+</pre>
+
+Returns an <code>iterator</code> that contains all of the token ids owned by the specified address.
+
+The parameter <code>owner</code> SHOULD be a 20-byte address.
+If not, this method SHOULD <code>throw</code> an exception.
+
+===Non-divisible NFT methods===
+
+====transfer====
+
+<pre>
+public static bool transfer(byte[] to, byte[] tokenid)
+</pre>
+
+It transfers the token with id <code>tokenid</code> to address <code>to</code>.
+
+The parameter <code>tokenid</code> SHOULD be a valid NFT.
+If not, this method SHOULD <code>throw</code> an exception.
+
+The parameter <code>to</code> SHOULD be a 20-byte address.
+If not, this method SHOULD <code>throw</code> an exception.
+
+The function SHOULD return <code>false</code> if the token that will be transferred has more than one owner.
```

Comment:
Do we really support multiple owners? And should we support them? I think one owner is enough (and that's what we have in the implementation), multisig scripts can be used if/when needed.

Reply:
This is different from multi-signature. The share held by each owner can be different.

Reply:
I agree with @erikzhang . There are different use cases for multiple owners and multi-sig addresses.

Reply:
Well, this `transfer` is in the "non-divisible" section and if we're to support multi-owned non-divisible NFTs some other interfaces might need to be corrected also (like `ownerOf` returning an array or iterator of owners).

Reply:
Non-divisible NFT couldn't be multi-owned.

Reply:
Then this sentence should be removed as non-divisible NFT can't have more than one owner.

Reply:
I moved it to `Common methods`.

Reply:
Divisible NFTs are supposed to implement this variant of `transfer` (as overloaded method) moving 100% of the token if there is only one owner of this token, correct?

Reply:
Yes.